src/mds/OpenFileTable.cc

```diff
@@ -342,8 +342,15 @@ void OpenFileTable::commit(MDSInternalContextBase *c, uint64_t log_seq, int op_p
     mds->objecter->mutate(oid, oloc, op, snapc, ceph::real_clock::now(), 0,
 			  gather.new_sub());
 
+#if !defined(__FreeBSD__)
```

Comment:
as discussed, this is not a limitation of FreeBSD, we should check for the feature of used C++ standard library in cmake by compiling a short C++ code using `map::merge()`. and define a macro in `config-h.in.cmake`, then check that macro here.

Reply:
i can help on this, if you are not familiar with cmake.

Reply:
@tchaikov 
That would be appriciated, I've figured out quite a bit of Cmake. But this would be new.